src/esp/assets/ResourceManager.h

```diff
@@ -84,6 +84,8 @@ class Model;
 }
 }  // namespace io
 namespace assets {
+// used for shadertype specification
+typedef metadata::attributes::ObjectInstanceShaderType ShaderTypeEnum;
```

Comment:
Do we really need the alias? it's named completely differently so it bears no resemblance to the type it aliases, there's a superfluous `Enum` suffix...

I feel we suffer from namespace overuse a bit, but introducing the same enum under a completely different name is only going to make it worse, not better in my opinion. Maybe just an `using` then?

```suggestion
using metadata::attributes::ObjectInstanceShaderType;
```

Reply:
That's cool.